src/types/onyx/OriginalMessage.ts

```diff
@@ -29,329 +36,600 @@ type OriginalMessageActionName =
     | 'ACTIONABLETRACKEXPENSEWHISPER'
     | 'TRIPPREVIEW'
     | ValueOf<typeof CONST.REPORT.ACTIONS.TYPE.POLICY_CHANGE_LOG>;
+
+/** Model of `approved` report action */
 type OriginalMessageApproved = {
+    /** Approved */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.APPROVED;
-    originalMessage: unknown;
+
+    /** Content of the original message */
+    originalMessage: {
+        /** Approved expense amount */
+        amount: number;
+
+        /** Currency of the approved expense amount */
+        currency: string;
+
+        /** Report ID of the expense */
+        expenseReportID: string;
+    };
 };
+
+/** Types of sources of original message */
 type OriginalMessageSource = 'Chronos' | 'email' | 'ios' | 'android' | 'web' | '';
 
+/** Model of `hold` report action */
 type OriginalMessageHold = {
+    /** Hold */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.HOLD;
+
+    /** Content of the original message */
     originalMessage: unknown;
 };
 
+/** Model of `hold comment` report action */
 type OriginalMessageHoldComment = {
+    /** Hold comment */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.HOLD_COMMENT;
+
+    /** Content of the original message */
     originalMessage: unknown;
 };
 
+/** Model of `unhold` report action */
 type OriginalMessageUnHold = {
+    /** Unhold */
     actionName: typeof CONST.REPORT.ACTIONS.TYPE.UNHOLD;
+
+    /** Content of the original message */
     originalMessage: unknown;
 };
 
+/** Details provided when sending money */
 type IOUDetails = {
+    /** How much was sent */
     amount: number;
+
+    /** Optional comment */
     comment?: string;
```

Comment:
NAB: I think the comment is optional so it can be an empty string but the property would be returned always
```suggestion
    comment: string;
```